posix_test.go

```diff
@@ -401,6 +404,167 @@ func TestDeleteFile(t *testing.T) {
+			0, 5, []byte("hello"),
+			func() error {
+				// On darwin HFS does not support > 1024 characters.
+				if runtime.GOOS == "darwin" {
+					return errFileNameTooLong
+				}
+				// On all other platforms return success.
+				return nil
+			}(),
+		},
+		// Object is a directory. - 5
+		{
+			"object-as-dir",
+			0, 5, nil, errIsNotRegular},
+		// One path segment length is > 255 chars long. - 6
+		{
+			"path/to/my/object0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001",
+			0, 5, nil, errFileNameTooLong},
+		// Path length is > 1024 chars long. - 7
+		{
+			"level0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001/level0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002/level0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000003/object000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001",
+			0, 5, nil, errFileNameTooLong},
+		// Buffer size greater than object size. - 8
+		{
+			"myobject", 0, 16, nil,
+			io.ErrUnexpectedEOF,
+		},
+		// Reading from an offset success. - 9
+		{
+			"myobject", 7, 5,
+			[]byte("world"), nil,
+		},
+		// Reading from an object but buffer size greater. - 10
+		{
+			"myobject",
+			7, 8,
+			[]byte("world"),
+			io.ErrUnexpectedEOF,
```

Comment:
This is a special case wherein both error and the result is returned, not sure how to validate for both, may be as a separate case since its very crucial case. CopyBuffer -> Hashsum -> IsValidBlock -> BitRotverify -> ReadParallel -> GetOBject -> , the entire execution tree will be influenced by this particular case. Will handle this copyBuffer tests too.